diff --git a/feeds/packages/libs/boost/Makefile b/feeds/packages/libs/boost/Makefile
index 25ceb8e..b10b352 100644
--- a/feeds/packages/libs/boost/Makefile
+++ b/feeds/packages/libs/boost/Makefile
@@ -168,11 +168,11 @@ define Package/boost/config
 				bool "Shared"

 			config boost-runtime-static
-				depends on @(!boost-shared-libs&&!boost-static-and-shared-libs)
+				depends on !boost-shared-libs&&!boost-static-and-shared-libs
 				bool "Static"

 			config boost-runtime-static-and-shared
-				depends on @(boost-use-name-tags&&!boost-shared-libs&&!boost-static-and-shared-libs)
+				depends on boost-use-name-tags&&!boost-shared-libs&&!boost-static-and-shared-libs
 				bool "Both"
 		endchoice

@@ -209,14 +209,14 @@ define Package/boost/config
 			default n

 		config boost-single-thread
-			depends on @boost-use-name-tags
+			depends on boost-use-name-tags
 			bool "Single thread Support."
 			help
 				Compile Boost libraries in single-thread mode.
 			default n

 		config boost-build-type-complete
-			depends on @boost-use-name-tags
+			depends on boost-use-name-tags
 			bool "Complete Boost Build."
 			help
 				Builds both release and debug libs. It will take much longer to compile.
