commit cdd2c4947e020b35e1cc51e24f5039c7c18193b5
Author: Sungjune Lee <sungjune@innomatic.ca>
Date:   Sat Apr 25 10:47:07 2020 +0900

    onion omega device patch

diff --git a/target/linux/ramips/base-files/etc/board.d/02_network b/target/linux/ramips/base-files/etc/board.d/02_network
index f743ce851a..aa49b09597 100755
--- a/target/linux/ramips/base-files/etc/board.d/02_network
+++ b/target/linux/ramips/base-files/etc/board.d/02_network
@@ -60,6 +60,7 @@ ramips_setup_interfaces()
 	ncs601w|\
 	omega2 | \
 	omega2p | \
+	omega2pro | \
 	timecloud|\
 	tplink,tl-wa801nd-v5|\
 	w150m|\
@@ -628,6 +629,7 @@ ramips_setup_macs()
 		lan_mac=$(macaddr_add "$lan_mac" 2)
 		;;
 	omega2|\
+	omega2pro | \
 	omega2p)
 		wan_mac=$(mtd_get_mac_binary factory 4)
 		lan_mac=$(mtd_get_mac_binary factory 46)
diff --git a/target/linux/ramips/base-files/lib/ramips.sh b/target/linux/ramips/base-files/lib/ramips.sh
index 093303892c..ebad591ce8 100755
--- a/target/linux/ramips/base-files/lib/ramips.sh
+++ b/target/linux/ramips/base-files/lib/ramips.sh
@@ -376,6 +376,9 @@ ramips_board_detect() {
 	*"Onion Omega2+")
 		name="omega2p"
 		;;
+	*"Onion Omega2 Pro")
+		name="omega2pro"
+		;;
 	*"OY-0001")
 		name="oy-0001"
 		;;
diff --git a/target/linux/ramips/dts/OMEGA2.dtsi b/target/linux/ramips/dts/OMEGA2.dtsi
index e36602d6c9..bbb181b474 100644
--- a/target/linux/ramips/dts/OMEGA2.dtsi
+++ b/target/linux/ramips/dts/OMEGA2.dtsi
@@ -6,13 +6,6 @@
 / {
 	compatible = "onion,omega2", "mediatek,mt7628an-soc";
 
-	aliases {
-		led-boot = &system_led;
-		led-failsafe = &system_led;
-		led-running = &system_led;
-		led-upgrade = &system_led;
-	};
-
 	chosen {
 		bootargs = "console=ttyS0,115200";
 	};
@@ -109,6 +102,7 @@
 		compatible = "jedec,spi-nor";
 		reg = <0>;
 		spi-max-frequency = <40000000>;
+		m25p,chunked-io = <31>;
 
 		partitions {
 			compatible = "fixed-partitions";
diff --git a/target/linux/ramips/dts/OMEGA2PRO.dts b/target/linux/ramips/dts/OMEGA2PRO.dts
new file mode 100644
index 0000000000..d0b7d4cba1
--- /dev/null
+++ b/target/linux/ramips/dts/OMEGA2PRO.dts
@@ -0,0 +1,39 @@
+/dts-v1/;
+
+#include "OMEGA2.dtsi"
+
+/ {
+	model = "Onion Omega2 Pro";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x8000000>;
+	};
+
+	new-leds {
+		compatible = "gpio-leds";
+
+		wifi {
+					label = "omega2pro:blue:wifi";
+					gpios = <&gpio1 12 GPIO_ACTIVE_LOW>;
+			};
+	};
+};
+
+&firmware {
+	reg = <0x50000 0xfb0000>;
+};
+
+&system_led {
+	label = "omega2pro:amber:system";
+	gpios = <&gpio1 11 GPIO_ACTIVE_LOW>;
+};
+
+&pinctrl {
+        state_default: pinctrl0 {
+                uart2 {
+			ralink,group = "uart2";
+                        ralink,function = "gpio";
+                };
+        };
+};
diff --git a/target/linux/ramips/image/mt76x8.mk b/target/linux/ramips/image/mt76x8.mk
index 7df5c04297..80cdf2f5f3 100644
--- a/target/linux/ramips/image/mt76x8.mk
+++ b/target/linux/ramips/image/mt76x8.mk
@@ -160,6 +160,14 @@ define Device/omega2p
 endef
 TARGET_DEVICES += omega2p
 
+define Device/omega2pro
+  DTS := OMEGA2PRO
+  IMAGE_SIZE := $(ralink_default_fw_size_16M)
+  DEVICE_TITLE := Onion Omega2 Pro
+  DEVICE_PACKAGES:= kmod-usb2 kmod-usb-ohci uboot-envtools kmod-sdhci-mt7620
+endef
+
+TARGET_DEVICES += omega2pro
 define Device/pbr-d1
   DTS := PBR-D1
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
